components/StoryList/index.js

```diff
@@ -1,52 +1,30 @@
-import React from "react";
-import styled from "styled-components";
 import Link from "next/link";
+import { useState, useEffect } from "react";
+import {
+  Header,
+  StoryListContainer,
+  StoryListCard,
+  StoryTitle,
+  StoryCoverImage,
+  StoryCreationDate,
+} from "./StyledStoryList";
 
-const Header = styled.h1`
-  font-size: 2rem;
-  text-align: center;
-`;
-const StoryListContainer = styled.div`
-  display: flex;
-  flex-direction: column;
-  flex-wrap: wrap;
-  justify-content: space-between;
-  align-content: center;
-  gap: 1rem;
-`;
-const StoryListCard = styled.div`
-  display: flex;
-  border: 1px solid #ccc;
-  border-radius: 4px;
-  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
-  padding: 1rem;
-  width: 90%;
+import initialStories from "@/public/db";
 
-  &:hover {
-    cursor: pointer;
-    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2);
-  }
-`;
+function StoryList() {
+  const [stories, setStories] = useState([]);
 
-const StoryTitle = styled.h2`
-  width: 100%;
-  font-size: 1.1rem;
-  margin: 0 0 0.5rem;
-`;
-const StoryCoverImage = styled.img`
-  width: 20%;
-  object-fit: cover;
-  margin-bottom: 0.5rem;
-`;
+  useEffect(() => {
+    const storedStories = JSON.parse(localStorage.getItem("stories"));
 
-const StoryCreationDate = styled.p`
-  align-self: self-end;
-  font-size: 0.8rem;
-  color: #666;
-  margin: 0;
-`;
+    if (storedStories && storedStories.length > 0) {
+      setStories(storedStories);
+    } else {
+      localStorage.setItem("stories", JSON.stringify(initialStories));
+      setStories(initialStories);
+    }
+  }, []);
```

Comment:
Instead of working with useEffect you can work with the local-storage-state-hook if you like :) See my comment above. Or you use the [useImmerLocalStorageState hook](https://github.com/mbosselmann/art-gallery-app/blob/main/lib/hook/useImmerLocalStorageState.js) that we introduced in recap project 5. :)